GooglePlayInstant/Editor/PlayInstantQuickDeployConfiguration.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+using UnityEngine;
+
+namespace GooglePlayInstant.Editor
+{
+    /// <summary>
+    /// Contains set of operations for storing and retrieving quick deploy configurations.
+    /// </summary>
+    public static class PlayInstantQuickDeployConfiguration
+    {
+        private static readonly string ConfigurationFilePath =
+            Path.Combine("Library", "PlayInstantQuickDeployConfig.json");
+
+        public static readonly Configuration Config = LoadConfiguration();
+
+        /// <summary>
+        /// Commit the current state of quick deploy configurations to persistent storage.
+        /// </summary>
+        public static void SaveConfiguration()
+        {
+            File.WriteAllText(ConfigurationFilePath, JsonUtility.ToJson(Config));
+        }
+
+        // Loads and returns Configuration representation of the configuration file if it exists, otherwise instantiate
+        // and return an empty Configuration instance.
+        private static Configuration LoadConfiguration()
+        {
+            if (!File.Exists(ConfigurationFilePath))
+            {
+                return new Configuration();
+            }
+
+            var configurationJson = File.ReadAllText(ConfigurationFilePath);
+            return JsonUtility.FromJson<Configuration>(configurationJson);
+        }
+
+        /// <summary>
+        /// Represents the contents of the quick deploy configuration file.
+        /// </summary>
+        [Serializable]
+        public class Configuration
+        {
+            public string assetBundleFileName;
+            public string cloudStorageBucketName;
+            public string cloudStorageFileName;
+            public string cloudCredentials;
+            public string assetBundleUrl;
+            public string loadingScreenImageFileName;
```

Comment:
I'm wondering if we need to save the filename or if we should just copy the file the user selects to a known location. Point being, the file needs to live in a certain location (under the Assets directory?) to be usable?

Reply:
This is used when a client uses the "Upload Loading Image" button, in which the tool then keeps track of the path the client specified. Using the "Create Loading Scene" button then creates a texture with the specified loading path and uses it as the Loading Scene's background. So as the project is currently, not sure if copying the file over is the solution as is.

I've two ideas to resolve this:
1) Rename "Upload Loading Image" to "Choose Loading Image". (so it's more clear that no uploading is occurring just yet.) and continuing to keep track of the loading screen image file name.
2) Keep it the same, but change the place where the place where the texture is created to the "Upload Loading Image". 

Keeping the file name better allows the developer to see what image they were intending to put into the loading screen if they exit out of the Quick Deploy Window and come back, so ultimately not sure if removing this field is good.

Either way, this brings up a problem I've realized. The project is constantly creating texture assets every time the user creates a loading scene without destroying the previous. I've added a TODO to fix this.

Reply:
Resolved offline. Will discuss in later PR, added TODO's for now.